File: problems/p10/p10.py
"""
project euler problem 10
"""
def sieve_eratos(num):
    """
    modified sieve of eratosthenes algorithm, takes a sum along the way
    """
    array = [True for i in range(num)]

    i = 2
    sum = 0
    while i*i <= num:
        if array[i]:
            for j in range(i*i, num, i):
                array[j] = False
        i += 1

    for i in range(2, num):
        if array[i]:
            sum += i

    return sum


def main():
    print(sieve_eratos(2000000))

if __name__ == '__main__':
    main()

File: 10/p10.cpp
/*
project euler problem 10
*/
#include <iostream>

long sieveEratos(int num){
  /*
  modified sieve of eratosthenes algorithm, takes a sum along the way
  */
  bool* arr = new bool[num];

  for (size_t i = 0; i < num; i++) {
    arr[i] = true;
  }

  int i = 2;
  long sum = 0;

  while (i*i <= num) {
    if (arr[i]) {
      for (size_t j = i*i; j < num; j += i) {
        arr[j] = false;
      }
    }
    i++;
  }

  for (size_t i = 2; i < num; i++) {
    if (arr[i]) {
      sum += i;
    }
  }

  delete [] arr;
  return sum;
}

int main(int argc, char const *argv[]) {
  std::cout << sieveEratos(2000000) << "\n";
  return 0;
}
